db/crud.go

```diff
@@ -48,10 +48,36 @@ func (db *DatabaseContext) GetDoc(docid string) (doc *document, err error) {
 	dbExpvars.Add("document_gets", 1)
 	if db.UseXattrs() {
 		var rawDoc, rawXattr []byte
-		cas, err := db.Bucket.GetWithXattr(key, KSyncXattrName, &rawDoc, &rawXattr)
-		doc, err = unmarshalDocumentWithXattr(docid, rawDoc, rawXattr, cas)
-		if err != nil {
-			return nil, err
+		importAttempts := 0
+		for {
+			cas, getErr := db.Bucket.GetWithXattr(key, KSyncXattrName, &rawDoc, &rawXattr)
+			if getErr != nil {
+				return nil, getErr
+			}
+			var unmarshalErr error
+			doc, unmarshalErr = unmarshalDocumentWithXattr(docid, rawDoc, rawXattr, cas)
+			if unmarshalErr != nil {
+				return nil, unmarshalErr
+			}
+			// If this was an SG Write, return the doc.  Otherwise, attempt to import
+			if doc.IsSGWrite() {
+				break
+			}
+
+			// Limit the number of import attempts for a given GET request, to guard against unexpected infinite looping
+			if importAttempts > 20 {
```

Comment:
The retry is triggered by a CAS collision, so it's different than our normal retry case (which is based on temporary server load spikes).  Given that, I don't think there's any value in a delay.